src/integration-tests/test_heartbeat.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2025 Bloomberg Finance L.P.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration test that suspends leader/replica/proxy/tool to trigger missed
+hearbeats followed by dead channel detection
+"""
+
+import blazingmq.dev.it.testconstants as tc
+from blazingmq.dev.it.fixtures import Cluster
+from blazingmq.dev.it.fixtures import (  # pylint: disable=unused-import
+    multi_node as cluster,
+    order,
+    tweak,
+)
+from blazingmq.dev.it.util import wait_until
+
+pytestmark = order(6)
+
+
+def _verify_delivery_to_new_consumer(proxy, uri, messages, timeout=2):
+    consumer = proxy.create_client("consumer")
+    consumer.open(uri, flags=["read"], succeed=True)
+    _verify_delivery(consumer, uri, ["msg1", "masg2"], timeout=2)
+    consumer.exit_gracefully()
+
+
+def _verify_delivery(consumer, uri, messages, timeout=2):
```

Comment:
This function could be moved to the top since it doesn't depend on anything in this file. Also `_verify_delivery_to_new_consumer` which is higher depends on this function.